File: backend/src/main/java/com/plugga/backend/entity/DeckCard.java
package com.plugga.backend.entity;

import javax.persistence.*;

@Entity
@Table(name = "deck_card")
public class DeckCard {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "deck_card_id")
    private int id;

    @ManyToOne
    @JoinColumn(name = "deck_id", referencedColumnName = "deck_id")
    private Deck deck;

    @ManyToOne
    @JoinColumn(name = "card_id", referencedColumnName = "card_id")
    private Card card;

    @OneToOne
    @JoinColumn(name = "pile_id", referencedColumnName = "pile_id")
    private Pile pile;

    public DeckCard() {
    }

    public int getId() {
        return id;
    }

    public void setId(final int id) {
        this.id = id;
    }

    public Deck getDeck() {
        return deck;
    }

    public void setDeck(final Deck deck) {
        this.deck = deck;
    }

    public Card getCard() {
        return card;
    }

    public void setCard(final Card card) {
        this.card = card;
    }

    public Pile getPile() {
        return pile;
    }

    public void setPile(final Pile pile) {
        this.pile = pile;
    }

    @Override
    public String toString() {
        return "DeckCard{" +
                "id=" + id +
                ", deck=" + deck +
                ", card=" + card +
                ", pile=" + pile +
                '}';
    }
}

File: backend/src/main/kotlin/com/plugga/backend/entity/DeckCard.kt
package com.plugga.backend.entity

import com.fasterxml.jackson.annotation.JsonIdentityReference
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.JoinColumn
import javax.persistence.ManyToOne
import javax.persistence.OneToOne
import javax.persistence.Table

@Entity
@Table(name = "deck_card")
class DeckCard {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "deck_card_id")
    var id: Int = 0

    @ManyToOne
    @JoinColumn(name = "deck_id", referencedColumnName = "deck_id")
    @JsonIdentityReference(alwaysAsId = true)
    var deck: Deck? = null

    @ManyToOne
    @JoinColumn(name = "card_id", referencedColumnName = "card_id")
    @JsonIdentityReference(alwaysAsId = true)
    var card: Card? = null

    @OneToOne
    @JoinColumn(name = "pile_id", referencedColumnName = "pile_id")
    @JsonIdentityReference(alwaysAsId = true)
    var pile: Pile? = null

    constructor() {}

    constructor(id: Int) {
        this.id = id
    }

    override fun toString(): String {
        return "DeckCard{" +
                "id=" + id +
                ", deck=" + deck +
                ", card=" + card +
                ", pile=" + pile +
                '}'.toString()
    }
}
